UnitTest/TestScripts/timingTest.cpp

```diff
@@ -0,0 +1,110 @@
+/* timingTest.cpp tests the timing module non-void functions
+ * Usage:
+ *  part of NS2 unit testing suite
+ *  to be called in unitTestDriver.cpp
+ *
+ */
+
+
+// NS2 includes
+#include "../headers/config.hpp"
+#include "../headers/timing.hpp"
+
+/* - - - - - - testWrapBufferIdx - - - - - - *
+ * Usage:
+ * tests wrapBufferIdx with variety of idx
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  0 - pass
+ *  1 - fail
+ */
+int testWrapBufferIdx() {
+
+    // build test cases and answers
+    int key[] = {0, BUFFERSIZE-1, BUFFERSIZE-1, 0, 1, 0};
+    int testVal[] = {0, -1, BUFFERSIZE-1, BUFFERSIZE, 2*BUFFERSIZE + 1, -BUFFERSIZE};
+
+    // call function to be tested
+    for (int i=0; i<6; i++) {
+        if (key[i] != wrapBufferIdx(testVal[i])) {
+            Serial.println("wrapBufferIdx() unit test failed (timing module)");
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+
+/* - - - - - - testVoltageRunningMean - - - - - - *
+ * Usage:
+ * tests voltageRunningMean with variety of buffers
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  0 - pass
+ *  1 - fail
+ */
+int testVoltageRunningMean() {
+
+    // build test cases and answers
+    float key[] = {0.200F, 0, 4.000F, 0, 16.000F};
+    int testIdx[] = {0, BUFFERSIZE-2*SMOOTH_IDX_COUNT, 2*SMOOTH_IDX_COUNT, 
+                      BUFFERSIZE-2*SMOOTH_IDX_COUNT, 2*SMOOTH_IDX_COUNT};

```

Comment:
why would the key have different values for the same values of `testIdx`? I'm not sure I'm understanding the logic here.

Reply:
I am doubling the buffer inside the test as mentioned in the above comment, so that is what leads to the key change